tests/test_flows.py

```diff
@@ -507,6 +507,28 @@ def test_channel_selection(self):
 
             env.checkpoint()
 
+        debug("Send invalid channel selection request to radio 0")
+        cs_req_mid = env.controller.dev_send_1905(
+            env.agents[0].mac, 0x8006,
+            # Single operating class with a single channel that doesn't exist in it.
+            tlv(0x8B, 0x000B, env.agents[0].radios[0].mac + ' 0x01 {0x52 {0x01 {0x01}} 0x00}'))
+        time.sleep(1)
+
+        cs_resp = self.check_cmdu_type("channel selection response", 0x8007, env.agents[0].mac,
+                                       env.controller.mac, cs_req_mid)
+
+        # TODO should be a single response (currently two are sent)
+        for response in cs_resp:
+            cs_resp_tlv = self.check_cmdu_has_tlv_single(response, 0x8e)
+            if cs_resp_tlv and \
+               cs_resp_tlv.channel_select_radio_id == env.agents[0].radios[0].mac:
+                if int(cs_resp_tlv.channel_select_response_code, 16) != 0x02:
```

Comment:
Use another 'and' condition instead of another 'if'

Reply:
I find it clearer to separate the actual assertion condition from the condition(s) that verify the context. But it's a thin line.

Anyway, this code will either go away or change in #1383 so I'm not going to bother fixing it.